src/sentry/api/endpoints/project_details.py

```diff
@@ -264,9 +264,18 @@ def validate_relayPiiConfig(self, value):
         return validate_pii_config_update(organization, value)
 
     def validate_relayCustomMetricCardinalityLimit(self, value):
-        if value is not None and value < 0:
+        if value is None:
+            return value
+
+        if value < 0:
             raise serializers.ValidationError("Cardinality limit must be a non-negative integer.")
 
+        # Value is stored as uint32 in relay
+        if value > 4_000_000_000:
```

Comment:
Why are we hardcoding this number here? Seems to be a place where divergence can happen very easily. Can't we validate the config entirely via the `sentry-relay` function in a place where we have access to the actual configuration we want to store?